pkg/query-service/app/integrations/messagingQueues/kafka/consumerLag.md

```diff
@@ -0,0 +1,197 @@
+## Consumer Lag feature break down
+
+### 1) Consumer Lag Graph
+
+
+---
+
+### 2) Consumer Group Details
+
+API endpoint:
+
+```
+POST /api/v1/messaging-queues/kafka/consumer-lag/consumer-details
+```
+
+```json
+{
+	"start": 1720685296000000000,
+	"end": 1721290096000000000,
+	"variables": {
+        "partition": "0",
+		"topic": "topic1",
+        "consumer_group": "cg1"
```

Comment:
How does the frontend fetch the values for `partition`, `topic` and `consumer_group`? Using `autocomplete/attribute_values`?

Reply:
Yes, basically in the design there's a dropdown for each of them, the values of which can be fetched via autocomplete APIs, once selected, those can be passed in this and the other messaging queue APIs.

Reply:
I had played around with `autocomplete/attribute_values` and it works for the partition, topic, and consumer_group
eg
the get API 
```bash
api/v3/autocomplete/attribute_values?aggregateOperator=avg&dataSource=traces&aggregateAttribute=kafka_consumer_group_lag&attributeKey=group&searchText=&filterAttributeKeyDataType=string&tagType=tag
```
would return all the consumer groups like this
```json
{
    "status": "success",
    "data": {
        "stringAttributeValues": [
            "cg4",
            "cg1",
            "cg2",
            "cg3"
        ],
        "numberAttributeValues": null,
        "boolAttributeValues": null
    }
}
```